kratos/python/add_mesh_to_python.cpp

```diff
@@ -374,6 +374,113 @@ pybind11::list GetValuesOnIntegrationPointsMatrix( TObject& dummy,
+    dummy.GetValueOnIntegrationPoints(rVariable, values, rCurrentProcessInfo);
+    for (size_t i = 0; i < values.size(); ++i)
+    {
+        pybind11::list integration_point_value;
+        integration_point_value.append(bool(values[i]));
+        values_list.append(integration_point_value);
+    }
+    return(values_list);
+}
+
+template< class TObject >
+pybind11::list GetValueOnIntegrationPointsDouble(
+    TObject& dummy,
+    const Variable<double>& rVariable,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    pybind11::list values_list;
+    size_t nb_integration_points = dummy.GetGeometry().IntegrationPointsNumber();
+    std::vector<double> values(nb_integration_points);
+    dummy.GetValueOnIntegrationPoints(rVariable, values, rCurrentProcessInfo);
+    for (size_t i = 0; i < values.size(); ++i)
+    {
+        pybind11::list integration_point_value;
+        integration_point_value.append(values[i]);
+        values_list.append(integration_point_value);
+    }
+    return(values_list);
+}
+
+template< class TObject >
+pybind11::list GetValueOnIntegrationPointsArray1d(
+    TObject& dummy,
+    const Variable<array_1d<double, 3> >& rVariable,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    pybind11::list values_list;
+    size_t nb_integration_points = dummy.GetGeometry().IntegrationPointsNumber();
+    std::vector<array_1d<double, 3> > values(nb_integration_points);
+    dummy.GetValueOnIntegrationPoints(rVariable, values, rCurrentProcessInfo);
+    for (size_t i = 0; i < values.size(); ++i)
+    {
+        pybind11::list integration_point_value;
+        for (size_t j = 0; j < 3; ++j)
+            integration_point_value.append(values[i][j]);
+        values_list.append(integration_point_value);
+    }
+    return(values_list);
+}
+
+template< class TObject >
+pybind11::list GetValueOnIntegrationPointsVector(
+    TObject& dummy,
+    const Variable<Vector>& rVariable,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    pybind11::list values_list;
+    size_t nb_integration_points = dummy.GetGeometry().IntegrationPointsNumber();
+    std::vector<Vector> values(nb_integration_points);
+    dummy.GetValueOnIntegrationPoints(rVariable, values, rCurrentProcessInfo);
+    for (size_t i = 0; i < values.size(); ++i)
+    {
+        pybind11::list integration_point_value;
+        for (size_t j = 0; j < values[i].size(); ++j)
+            integration_point_value.append(values[i][j]);
+        values_list.append(integration_point_value);
+    }
+    return(values_list);
+}
+
+template< class TObject >
+pybind11::list GetValueOnIntegrationPointsMatrix(
+    TObject& dummy,
+    const Variable<Matrix>& rVariable,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    pybind11::list values_list;
+    size_t nb_integration_points = dummy.GetGeometry().IntegrationPointsNumber();
+    std::vector<Matrix> values(nb_integration_points);
+    dummy.GetValueOnIntegrationPoints(rVariable, values, rCurrentProcessInfo);
```

Comment:
this should be mapped to `CalculateOnIntegrationPoints` I would say, its what we wanted to do in #2194